### PR TITLE
Build VS redist for ANCM

### PR DESCRIPTION
**Pull request by jkotalik**
Right now, we insert ANCM via a convoluted method with creating a nupkg via an internal feed. This attempts to create a nuget package for ANCM for IIS and IIS Express.

This needs more work, but this is for testing on CI that the nupkgs created are correct. There is reactionary to actually insert these correctly into VS.

### CODE REVIEW

**Comment by jkotalik** [comment]
Should have marked this as draft, still want to wait to verify on CI these nupkgs are correct.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    <ProjectReference Include="ANCMIISExpressV2\AncmIISExpressV2.wixproj">
+      <SetPlatform>Platform=$(Platform)</SetPlatform>
+      <Name>AspNetCoreModuleIISExpressV2_$(Platform)</Name>
+      <Project>f9bacb48-3bd7-4ec2-ae31-664e8703ec12</Project>
+      <Private>True</Private>
+      <DoNotHarvest>true</DoNotHarvest>
+    </ProjectReference>
+  </ItemGroup>
+
+  <Target Name="CreateSharedFrameworkNugetPackage" AfterTargets="CopyToArtifactsDirectory;Build">
```
Comment (wtgodbe) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj:79]: Target name should be different

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    <DefineConstants>$(DefineConstants);ProductNameShort=$(ProductNameShort)</DefineConstants>
+    <DefineConstants>$(DefineConstants);AspNetCoreSharedFrameworkSource=$(HarvestSource)</DefineConstants>
+    <NamespaceGuid>$(SharedFrameworkNamespaceGuid)</NamespaceGuid>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ToolsetInstallerNuspecFile>$(RepoRoot)\src\Installers\Windows\AspNetCoreModule-Setup\ANCMVS.nuspec</ToolsetInstallerNuspecFile>
+    <SkipCopyToArtifactsDirectory>true</SkipCopyToArtifactsDirectory>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <WixExtension Include="WixDependencyExtension">
```
Comment (wtgodbe) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj:28]: Is all of the wix stuff needed? I thought we just wanted to repackage the ANCM .msi's/assets that get created elsewhere

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.nuspec [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        <icon>$PackageIcon$</icon>
+        <requireLicenseAcceptance>true</requireLicenseAcceptance>
+        <description>$MAJOR$.$MINOR$ ASP.NET Core Module ($ARCH$) Windows Installer MSI as a .nupkg for internal Visual Studio build consumption</description>
+        <copyright>© Microsoft Corporation. All rights reserved.</copyright>
+        <serviceable>true</serviceable>
+    </metadata>
+    <files>
+        <file src="$ASPNETCORE_RUNTIME_MSI$" />
+        <!-- Reusing ASPNETCORE_CAB_FILE to add both IIS and IISExpress versions of ANCM. -->
+        <file src="$ASPNETCORE_CAB_FILE$" />
```
Comment (joeloff) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.nuspec:20]: There's a CAB file? ANCM has always had embedded CABs (ASP.NET had external CABs long ago in 2.0)

**Comment by jkotalik** [comment]
@dotnet/aspnet-build is there still no way to build installers as part of the public build? Do I need to make a branch internally to verify their creation?

**Comment by wtgodbe** [comment]
@jkotalik you can add a block like the following to the `artifacts:` blocks for winx64/x86 (or wherever else you want):

https://github.com/dotnet/aspnetcore/blob/0e195da775478fd49b5c476dc333c99a115ad9ce/.azure/pipelines/ci.yml#L287-L288

This branch did already upload your new VS.Redist packages, you can find them in the `nonshipping` section of the winx64/x86 package artifacts

**Comment by jkotalik** [comment]
Cool, this generally seemed like it worked!

Thanks will, I didn't realize that those packages would be present when the installers weren't.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.nuspec (hunk 3 above)
Comment (jkotalik) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.nuspec:20]: I've overloading this script to use the the ASPNETCORE_CAB_FILE as the other msi. There are two msis I'd like to insert into this package, and the script only accepted one msi.

I can probably clean this up to use a different variable, but this was easier.

File: src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    <OutputNamePrefix>aspnetcoremodulevs-</OutputNamePrefix>
+    <OutputNameSuffix>-win-$(Platform)</OutputNameSuffix>
+    <ProductNameFolder>Microsoft ASP.NET Core Module</ProductNameFolder>
+    <ProductNameShort>AspNetCoreModule.VS</ProductNameShort>
+    <OutputType>Package</OutputType>
+    <ProjectGuid>541E1EF7-E129-4274-A05A-321F785AA2E9</ProjectGuid>
+    <IsShipping>true</IsShipping>
+    <DefineConstants>$(DefineConstants);ProductNameFolder=$(ProductNameFolder)</DefineConstants>
+    <DefineConstants>$(DefineConstants);ProductNameShort=$(ProductNameShort)</DefineConstants>
+    <NamespaceGuid>$(ProjectGuid)</NamespaceGuid>
```
Comment (jkotalik) [inline src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj:15]: @joeloff do I need  to do anything specific with these GUIDs?

Comment (joeloff) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj:15]: no, the namespace guid would only be used to generate things like the ProjectCode/UpgradeCode (assuming this project uses the same task the runtime/targeting packs use)

Comment (jkotalik) [reply src/Installers/Windows/AspNetCoreModule-Setup/ANCMVS.wixproj:15]: I think they use the same tasks, I hit an error when I didn't have a guid.

**Comment by jkotalik** [comment]
@leecow we were hoping to see if we could add another two packages into our mechanism for inserting dotnet into VS. These two would include the ASP.NET Core Module, which is used with IIS + ASP.NET Core.

This PR will create two packages (x64 and x86), but what other work is needed to actually get the insertion flow working into VS?

**Comment by leecow** [comment]
Adding @marcpopMSFT and @wli3 to help with the specifics as their team handles most of the VS insertion mechanics.

**Comment by jkotalik** [comment]
I've reach out to them off of github. Making progress on this but needs a few updates.

**Comment by jkotalik** [comment]
Closing for now as I haven't made progress on this.

**Pull request closed without merging**